tests/Command/TestUsersMigrator.php

```diff
@@ -0,0 +1,452 @@
+		foreach ( $author_ids as $author_id ) {
+			$post_ids[] = self::factory()->post->create(
+				[
+					'post_author' => $author_id,
+					'post_type'   => 'post',
+					'post_status' => 'publish',
+				]
+			);
+		}
+
+		// Delete all users except editors and administrators
+		UsersMigrator::cmd_delete_users_by_role(
+			[],
+			[
+				'exclude-roles' => 'editor,administrator',
+			]
+		);
+
+		// Assert that subscribers and authors were deleted
+		$subscribers = get_users( [ 'role' => 'subscriber' ] );
+		$authors     = get_users( [ 'role' => 'author' ] );
+		$this->assertEquals( 0, count( $subscribers ) );
+		$this->assertEquals( 0, count( $authors ) );
+
+		// Assert that editors and administrators were preserved
+		$editors = get_users( [ 'role' => 'editor' ] );
+		$admins  = get_users(
+			[
+				'role'    => 'administrator',
+				'exclude' => [ 1 ], // Exclude the default WordPress admin user
+			]
+		);
+		$this->assertEquals( 2, count( $editors ), 'Wrong number of editors' );
+		$this->assertEquals(
+			3,
+			count( $admins ),
+			sprintf(
+				'Wrong number of administrators (excluding default admin). Found: %s',
+				wp_json_encode(
+					array_map(
+						function( $user ) {
+							return [
+								'ID'         => $user->ID,
+								'user_login' => $user->user_login,
+							];
+						},
+						$admins
+					)
+				)
+			)
+		);
+
+		// Debug information
+		$admin_info = array_map(
+			function( $user ) {
+				return [
+					'ID'         => $user->ID,
+					'user_login' => $user->user_login,
+				];
+			},
+			$admins
+		);
+		$this->assertEquals( 3, count( $admins ), sprintf( 'Wrong number of administrators. Found: %s', wp_json_encode( $admin_info ) ) );
+
+		// Verify posts were deleted
+		foreach ( $post_ids as $post_id ) {
+			$this->assertNull( get_post( $post_id ) );
+		}
+	}
+
+	/**
+	 * Test that roles and exclude-roles cannot be used together.
+	 */
+	public function test_roles_and_exclude_roles_mutually_exclusive(): void {
+		// Create a test user
+		$user_id = self::factory()->user->create( [ 'role' => 'author' ] );
+
+		// Try to use both parameters
+		UsersMigrator::cmd_delete_users_by_role(
+			[],
+			[
+				'roles'         => 'author',
+				'exclude-roles' => 'administrator',
+			]
+		);
+
+		// Assert that no users were deleted
+		$authors = get_users( [ 'role' => 'author' ] );
+		$this->assertEquals( 1, count( $authors ) );
+	}
+
+	/**
+	 * Test that at least one of roles or exclude-roles must be provided.
+	 */
+	public function test_roles_or_exclude_roles_required(): void {
+		// Create a test user
+		$user_id = self::factory()->user->create( [ 'role' => 'author' ] );
+
+		// Try to run command without either parameter
+		UsersMigrator::cmd_delete_users_by_role(
+			[],
+			[]
+		);
+
+		// Assert that no users were deleted
+		$authors = get_users( [ 'role' => 'author' ] );
+		$this->assertEquals( 1, count( $authors ) );
+	}
+
+	/**
+	 * Clean up after all tests.
+	 */
+	public function tearDown(): void {
```

Comment:
Does all this (maybe except for the wp hook/removal) not happen automatically? I've never had to tear down stuff like this, but I could be wrong.

Reply:
Good point. I deleted the tearDown method.